tests/test_mathematical.py

```diff
@@ -4168,25 +4168,85 @@ def test_matmul_with_offsets(self, sh1, sh2):
         assert_array_equal(result, expected)
 
 
+class TestMatmulInplace:
+    ALL_DTYPES = get_all_dtypes(no_none=True)
+    DTYPES = {}
+    for i in ALL_DTYPES:
+        for j in ALL_DTYPES:
+            if numpy.can_cast(j, i):
+                DTYPES[f"{i}-{j}"] = (i, j)
+
+    @pytest.mark.parametrize("dtype1, dtype2", DTYPES.values())
+    def test_basic(self, dtype1, dtype2):
+        a = numpy.arange(10).reshape(5, 2).astype(dtype1)
+        b = numpy.ones((2, 2), dtype=dtype2)
+        ia, ib = dpnp.array(a), dpnp.array(b)
+        ia_id = id(ia)
+
+        a @= b
+        ia @= ib
+        assert id(ia) == ia_id
+        assert_dtype_allclose(ia, a)
+
+    @pytest.mark.parametrize(
+        "a_sh, b_sh",
+        [
+            pytest.param((10**5, 10), (10, 10), id="2d_large"),
+            pytest.param((10**4, 10, 10), (1, 10, 10), id="3d_large"),
+            pytest.param((3,), (3,), id="1d"),
+            pytest.param((3, 3), (3,), id="2d_1d"),
```

Comment:
For such a case `NumPy` raises an Error with a more informative msg. Should we align `dpnp` to behave similarly?

```python
import numpy
a=numpy.ones((4, 3))
b=numpy.ones(3)
a @= b
# ValueError: inplace matrix multiplication requires the first operand to have at least one and the second at least two dimensions.

import dpnp
a=dpnp.ones((4, 3))
b=dpnp.ones(3)
a @= b
# ValueError: Axes item 1 should be a tuple with a single element, or an integer.
```

Reply:
Done